alibi_detect/utils/loading.py

```diff
@@ -0,0 +1,625 @@
+
+    Returns
+    -------
+    The preprocess_fn function.
+    """
+    preprocess_fn = cfg.pop('src')
+
+    if callable(preprocess_fn):
+        if preprocess_fn.__name__ == 'preprocess_drift':
+            # If preprocess_drift function, kwargs is preprocess cfg minus 'function' and 'kwargs'
+            cfg.pop('kwargs')
+            kwargs = cfg.copy()
+
+            # Final processing of model (and/or embedding)
+            model = kwargs['model']
+            emb = kwargs.pop('embedding')  # embedding passed to preprocess_drift as `model` therefore remove
+
+            # Backend specifics
+            if backend == 'tensorflow':
+                model = prep_model_and_emb_tf(model, emb)
+                kwargs.pop('device')
+            elif backend == 'pytorch':  # TODO - once optional deps implemented
+                raise NotImplementedError('Loading preprocess_fn for PyTorch not yet supported.')
+#               device = cfg['device'] # TODO - device should be set already - check
+#               kwargs.update({'model': kwargs['model'].to(device)})  # TODO - need .to(device) here?
+#               kwargs.update({'device': device})
+            kwargs.update({'model': model})
+        else:
+            kwargs = cfg['kwargs']  # If generic callable, kwargs is cfg['kwargs']
+
+    else:
+        logger.warning('Unable to process preprocess_fn. No preprocessing function is defined.')
+        return None
+
+    return partial(preprocess_fn, **kwargs)
+
+
+def _load_model_config(cfg: dict,
+                       backend: str) -> Callable:
+    """
+    Loads TensorFlow, PyTorch and scikit-learn models (currently only TensorFlow supported), from a model config
+    dict.
+
+    Parameters
+    ----------
+    cfg
+        Model config dict. (see pydantic model schemas).
+    backend
+        The backend.
+
+    Returns
+    -------
+    The loaded model.
+    """
+
+    # Load model
+    src = cfg['src']
+    custom_obj = cfg['custom_objects']
+    layer = cfg['layer']
+    src = Path(src)
+    if not src.is_dir():
+        raise FileNotFoundError("Compatible model not found at %s" % str(src.resolve()))
+
+    if backend == 'tensorflow':
+        model = load_model_tf(src, load_dir='.', custom_objects=custom_obj, layer=layer)
+    else:
+        raise NotImplementedError('Loading of pytorch models not currently supported')
+
+    return model
+
+
+def _load_embedding(cfg: dict, backend: str) -> Callable:  # TODO: Could type return more tightly
+    """
+    Load a pre-trained text embedding from an embedding config dict.
+
+    Parameters
+    ----------
+    cfg
+        An embedding config dict. (see the pydantic schemas).
+    backend
+        The backend.
+
+    Returns
+    -------
+    The loaded embedding.
+    """
+    src = cfg['src']
+    layers = cfg['layers']
+    typ = cfg['type']
+    if backend == 'tensorflow':
+        emb = load_embedding_tf(src, embedding_type=typ, layers=layers)
+    else:
+        emb = None
```

Comment:
Noting that the logic here is different than for loading the model, i.e. returning `None` instead of raising an error.